### PR TITLE
Clean up the utility codes

### PR DESCRIPTION
**Pull request by kezhenxu94**
Clean up the over-used utility codes and remove some third party codes to avoid license issues.

### CODE REVIEW

**Comment by kezhenxu94** [comment]
@CalvinKirs @lenboo @dailidong please take a look

**Review by CalvinKirs — DISMISSED** [review]
LGTM

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java [hunk 1]
```diff
@@ -300,7 +301,7 @@
             if (taskNodeList.size() != taskRelationList.size()) {
                 Set<Long> postTaskCodes = taskRelationList.stream().map(ProcessTaskRelationLog::getPostTaskCode).collect(Collectors.toSet());
                 Set<Long> taskNodeCodes = taskNodeList.stream().map(TaskNode::getCode).collect(Collectors.toSet());
-                Collection<Long> codes = CollectionUtils.subtract(postTaskCodes, taskNodeCodes);
+                Collection<Long> codes = org.apache.commons.collections4.CollectionUtils.subtract(postTaskCodes, taskNodeCodes);
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java]: Is seems we can only use `CollectionUtils.subtract(postTaskCodes, taskNodeCodes)` here?

Comment (kezhenxu94) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java]: Yes, changed the other methods to `collections4`

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Pull request merged**